boards: renesas: rx: Add support for RX261 boards

quytranpzz opened the pull request:
- Add SOC support for Renesas RX261 MCU
- Add support for EK-RX261 and FPB-RX261 boards
- Add scripts to support flash using RFP (Renesas flash programmer)
- Update serial driver, using a common sci struct which already defined in HAL

Commit:
soc: renesas: rx: Initial support for RX261 SOC

This commit to initial support for RX261 SOC.

Signed-off-by: Hau Ho <hau.ho.xc@bp.renesas.com>
Signed-off-by: Phi Tran <phi.tran.jg@bp.renesas.com>

Commit:
dts: renesas: initial support dts SoC layer on RX261.

This commit to initial support dts SoC layer on RX261

Signed-off-by: Hau Ho <hau.ho.xc@bp.renesas.com>
Signed-off-by: Phi Tran <phi.tran.jg@bp.renesas.com>

Commit:
boards: renesas: rx: Initial support for EK-RX261 board

Initial support for EK-RX261 board using RX261 SOC

Singed-off-by: Quy Tran <quy.tran.pz@renesas.com>
Signed-off-by: Hau Ho <hau.ho.xc@bp.renesas.com>

Commit:
drivers: clock control: Add support clock control for RX261

Add support clock control for RX261

Signed-off-by: Phi Tran <phi.tran.jg@bp.renesas.com>

Commit:
drivers: gpio: Update gpio and pinctrl driver for support RX261

Update gpio driver and pinctrl driver for support RX261

Signed-off-by: Phi Tran <phi.tran.jg@bp.renesas.com>

Commit:
drivers: uart: Update sci uart for support RX261

This commit to update sci uart for support RX261

Signed-off-by: Phi Tran <phi.tran.jg@bp.renesas.com>

Commit:
boards: renesas: rx: Initial support for FPB-RX261 board

Initial support for board FPB-RX261

Signed-off-by: Phi Tran <phi.tran.jg@bp.renesas.com>

Commit:
arch: update fixed vectors for config HOCO frequency.

Add macro to update fixed vectors to configure Hoco frequency
via the register OFS1.

Signed-off-by: Phi Tran <phi.tran.jg@bp.renesas.com>

Commit:
scripts: west: flash: Add support for scripts to flash using the RFP tool.

Add support for scripts to flash using the RFP tool.

Signed-off-by: Hau Ho <hau.ho.xc@bp.renesas.com>

quytranpzz commented on boards/renesas/fpb_rx261/fpb_rx261_defconfig:
```diff
@@ -0,0 +1,13 @@
+# Enable GPIO
+CONFIG_GPIO=y
+
+# Enable uart driver
```
Updated

quytranpzz commented on boards/renesas/ek_rx261/ek_rx261.dts:
```diff
@@ -0,0 +1,98 @@
+	pinctrl-0 = <&sci6_default>;
+	pinctrl-names = "default";
+	status = "okay";
+	uart6: uart {
```
Added new line, thanks

dsseng commented on soc/renesas/rx/rx261/soc.c:
```diff
@@ -0,0 +1,48 @@
+	bsp_ram_initialize();
+	bsp_interrupt_open();
+	bsp_register_protect_open();
+#if CONFIG_RENESAS_NONE_USED_PORT_INIT == 1
```
```suggestion
#if CONFIG_RENESAS_NONE_USED_PORT_INIT
```

dsseng commented on soc/renesas/rx/rx261/Kconfig.soc:
```diff
@@ -0,0 +1,20 @@
+	bool
+	select SOC_SERIES_RX261
+	help
+	  SOC_R5F52618BGFP
```
```suggestion
	  R5F52618BGFP microcontroller Soc
```

dsseng commented on boards/renesas/ek_rx261/ek_rx261.dts:
```diff
@@ -0,0 +1,98 @@
+};
+
+&cmt {
+	clock-frequency = <4000000>;
```
Suggestion for consistency:

```suggestion
	clock-frequency = <DT_FREQ_M(4)>;
```

quytranpzz replied:
Updated, thanks

dsseng commented on dts/rx/renesas/r5f52618bgfp.dtsi:
```diff
@@ -0,0 +1,235 @@
+			#size-cells = <1>;
+			compatible = "renesas,rx-flash.yaml";
+			reg = <0x007e0000 0x1000>;
+			code_flash: flash@fff80000 {
```
Unsure if necessary

```suggestion

			code_flash: flash@fff80000 {
```

dsseng commented on dts/bindings/clock/renesas,rx-cgc-root-clock.yaml:
```diff
@@ -43,7 +43,13 @@ properties:
       Drive Capacity Control (for Sub-Clock Oscillator only)
       0: Drive capacity for standard CL.
-      1: Drive capacity for low CL.
+      1: Drive capacity for low CL. (in case of rx261, not choose this)
```
```suggestion
      1: Drive capacity for low CL. (in case of rx261, do not choose this)
```

or

```suggestion
      1: Drive capacity for low CL. (unsupported on rx261)
```

?

dsseng reviewed (commented):
some minor suggestions

quytranpzz commented:
Hello @nordicjm , @duynguyenxa , I just rebased main to include the HAL support, and added documentation for these boards. Could you please take a look? Thanks

duynguyenxa commented on soc/renesas/rx/rx261/ofsm.c:
@quytranpzz , this file was not added in the CMakeLists.txt, it's seem that is unnecessary for now, please remove it,   later we need to add a mapping mechanism for ofsm setting to the hal_layer

quytranpzz replied:
It's done

duynguyenxa commented on boards/renesas/fpb_rx261/fpb_rx261.dts:
```diff
@@ -0,0 +1,86 @@
+
+/ {
+	model = "Renesas FPB-RX261";
+	compatible = "renesas,fpb_rx261","renesas,rxv3";
```
@quytranpzz , please remove the `"renesas,rxv3"`

duynguyenxa commented on boards/renesas/ek_rx261/ek_rx261.dts:
```diff
@@ -0,0 +1,98 @@
+
+/ {
+	model = "Renesas EK-RX261 KIT";
+	compatible = "renesas,ek_rx261","renesas,rxv3";
```
@quytranpzz , please remove the `"renesas,rxv3"`

nordicjm reviewed (approved):
SoC/board files OK
